### ISSUE TITLE
How to add new metrics? 

### ISSUE THREAD

**Issue by jianlong-yuan** [open-mmlab/mmsegmentation#1286]
Could we register new metrics? When we modify new metrics, it is inflexible.

**Comment by MengzhangLI** [comment]
If the metric is not related with logits score, you can check this pr: https://github.com/open-mmlab/mmsegmentation/pull/509.

The logits score for metric (for example, AUC) needs refactoring of existed code, it is WIP.
